jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ArrayByteBufferPool.java

```diff
@@ -581,29 +588,69 @@ private Pool.Entry<RetainableByteBuffer> evict()
         }
     }
 
-    private static class Buffer extends AbstractRetainableByteBuffer
+    private interface Removable
```

Comment:
I'd prefer to make this interface an abstract class extending from `AbstractRetainableByteBuffer`.